js/pom.xml

```
@@ -51,6 +51,41 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-clean-plugin</artifactId>
+                <configuration>
+                    <followSymLinks>false</followSymLinks>
+                    <filesets>
+                        <fileset>
+                            <directory>.wireit</directory>
+                            <includes><include>.wireit</include></includes>
+                        </fileset>
+                        <fileset>
+                            <directory>node_modules</directory>
+                            <includes><include>**</include></includes>
+                        </fileset>
+                        <!-- include all non-maven projects here as well -->
+                        <fileset>
+                            <directory>libs</directory>
+                            <includes>
+                                <include>keycloak-masthead/node_modules/**</include>
+                                <include>keycloak-masthead/.wireit/**</include>
+                                <include>ui-shared/node_modules/**</include>
+                                <include>ui-shared/.wireit/**</include>
+                            </includes>
+                        </fileset>
+                        <fileset>
+                            <directory>apps/keycloak-server</directory>
+                            <includes>
+                                <include>server/**</include>
+                                <include>node_modules/**</include>
+                                <include>.wireit/**</include>
+                            </includes>
+                        </fileset>
+                    </filesets>
```

Perhaps we could recursively remove all `node_modules` directories, rather than list them all separately (untested):
```suggestion
                        <fileset>
                            <directory>.</directory>
                            <includes>
                                <include>**/.wireit/**</include>
                                <include>**/node_modules/**</include>
                            </includes>
                        </fileset>
                        <fileset>
                            <directory>apps/keycloak-server</directory>
                            <includes>
                                <include>server/**</include>
                            </includes>
                        </fileset>
                    </filesets>
```

---

Yup, that's certainly better. Good to know that any of these subdirectories can be removed.